roles/kvm/handlers/main.yml

```diff
@@ -0,0 +1,36 @@
+---
+- name: Verify KVM installation
+  block:
+    - name: Check if libvirtd service is running
+      ansible.builtin.service_facts:
+      register: service_state
+
+    - name: Assert libvirtd service is running
+      ansible.builtin.assert:
+        that:
+          - "'libvirtd.service' in service_state.ansible_facts.services"
+          - "service_state.ansible_facts.services['libvirtd.service'].state == 'running'"
+        fail_msg: "libvirtd service is not running"
+        success_msg: "libvirtd service is running correctly"
+
+    - name: Check if virsh command exists
+      ansible.builtin.command: which virsh
+      register: virsh_check
+      changed_when: false
+      failed_when: virsh_check.rc != 0
```

Comment:
I think that these failed_when are not necessary. because that should be the default behavior with command (failing with rc != 0). Could you try to remove them and check if everything works as expected?

Reply:
You are absolutely right here, the command will fail automatically without an explicit check. Removed the redundant checking.